src/main/java/zean/task/Deadline.java

```diff
@@ -0,0 +1,76 @@
+package zean.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import zean.exception.DukeException;
+
+/**
+ * The class representing a deadline task.
+ *
+ * @author Zhong Han
+ */
+public class Deadline extends Task {
+    private LocalDate by;
+
+    /**
+     * Constructor for the deadline task.
+     *
+     * @param description The description of the deadline task.
+     * @param by The due date of the deadline task.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description.strip());
+        try {
+            this.by = LocalDate.parse(by.strip());
+        } catch (DateTimeParseException e) {
+            throw new DukeException("The date is invalid!");
+        }
+    }
+
+    /**
+     * Constructor for the deadline task.
+     *
+     * @param bool The completion status of the deadline task.
+     * @param description The description of the deadline task.
+     * @param by The due date of the deadline task in the format YYYY-MM-DD.
+     */
+    public Deadline(String bool, String description, String by) {
+        super(description.strip());
+        this.by = LocalDate.parse(by.strip());
+        if (Integer.parseInt(bool) == 1) {
+            this.isDone = true;
+        }
+    }
```

Comment:
the logic for these two constructors are somewhat similar, are there way to combine them?